### ISSUE TITLE
MOTD Menu, String Expected

### ISSUE THREAD

**Issue by Violat0r** [Nayruden/Ulysses#470]
Here is the error:

[ERROR] addons/ulx/lua/ulx/modules/cl/motdmenu.lua:29: bad argument #1 to 'SetHTML' (string expected, got no value)
  1. SetHTML - [C]:-1
   2. fn - addons/ulx/lua/ulx/modules/cl/motdmenu.lua:29
    3. func - addons/ulib/lua/ulib/client/cl_util.lua:22
     4. unknown - lua/includes/extensions/net.lua:32

I should point out I am using ulx_motd.txt instead of a URL. However I am unable to add a url via menu. Of course the motd is not showing when players join. This is using the latest files off git.

I revered back to an older version of the ulx and ulib. I am able to put a url in now however I am still getting the same message using ulx_motd.txt.

**Comment by SticklyMan** [comment]
Basically, it looks like ULib is not able to find your ulx_motd.txt file- It should be located at \data\ulx_motd.txt, and you should make sure that the clients are receiving this file.

Hmm, how are you unable to set a URL via XGUI? Do you get any errors or is the textfield disabled?

**Comment by Violat0r** [comment]
Correct me if I am wrong, but isn't the path to the MOTD garrysmod/addons/ulx_motd.txt?

I was unable to reproduce this error on another server.

I was able to reproduce not being able to enter a URL. See http://i.imgur.com/x2opd37.jpg

**Comment by SticklyMan** [comment]
Hmm, I did just change something relating to how XGUI handles the MOTD changing- but I may have broke something. I'll take a look at it ASAP within the next couple days. If you can give me steps to reproduce, that'd be very helpful.

And looking at the code, the server reads ulx_motd.txt in the Garry's Mod root directory, which will read from garrysmod/addons/ulx/ulx_motd.txt if not in the root. (I'm not sure which one will take priority if both exist, actually)
https://github.com/Nayruden/Ulysses/blob/master/ulx/lua/ulx/modules/sh/menus.lua#L9

The client is sent the ulx_motd.txt file at some point, which is then stored in the Client's data/ulx_motd.txt. This is where your error is occurring- the client's file does not exist when it tries to display the MOTD.
https://github.com/Nayruden/Ulysses/blob/master/ulx/lua/ulx/modules/cl/motdmenu.lua#L29

**Comment by Violat0r** [comment]
I upgraded ulx and ulib from an older version, about 2 months old. I completely deleted the old folders and replaced them with the new folders. Once that was completed I launched the server via script on my local computer. After that I join the server the motd comes up and I close it. I type in !menu and go to the settings tab and select general settings. Show motd when player joins is unchecked, yet I am shown the motd when I join the server every time. Checking it makes no difference, the motd still comes up. However, when I now uncheck show motd when player joins and reconnect, I no longer see the motd show up. This is another bug that should be looked into. It might take a few tries but you will get the motd to show up even when its unchecked when you join the server. Anyways, the other bug the steps are the same by going to !menu and going to settings and then general settings. I am unable to check or uncheck the url setting. I cannot type in the box either. I have tried changing other settings under general settings to see if anything will unlock it with no luck. I tried this all while in spec and out of spec, made no difference. I also have all permissions checked.

**Comment by Violat0r** [comment]
Since we are on the subject of motds, I also have another bug I would like to point out. I am not sure if this is because of ulx or gmod. I have found if I set the motd to a url that contains a lot of scripts, ajax, and what not, it slows my computer. 

I join the sever, minimize the game for a few hours, then I begin to take notice my computer is slowing down. I reopen gmod and then minimize again and some of the icons in windows taskbar overlap. This doesn't go away until I close gmod. This doesn't happen when the motd is set to a simple html motd so I do believe there is some kind of "leak" going on when using sites with heavy programming on them and not closing the website for awhile. I am unsure if this is a gmod problem or the way ulx handles the website. I am on windows 8.1 using the latest nvidia drivers. If desired, I can provide the ip to the server and the site.
